src/main/java/ca/corefacility/bioinformatics/irida/ria/web/analysis/AnalysisController.java

```diff
@@ -151,4 +167,59 @@ public String getAdvancedPhylogeneticVisualizationPage(@PathVariable Long submis
 		model.addAttribute("submission", submission);
 		return BASE + "visualizations/phylocanvas-metadata";
 	}
+
+	/**
+	 * Get the html page from the file name provided.
+	 *
+	 * @param submissionId {@link Long} identifier for an {@link AnalysisSubmission}
+	 * @param filename     The html file name
+	 * @param locale       User's locale
+	 * @param response     {@link HttpServletResponse}
+	 * @throws IOException if we can't write the file to the response
+	 */
+	@RequestMapping("/{submissionId}/html-output")
+	public void getHtmlOutputForSubmission(@PathVariable Long submissionId, @RequestParam String filename,
+			Locale locale, HttpServletResponse response) throws IOException {
+		AnalysisSubmission submission = analysisSubmissionService.read(submissionId);
+		Set<AnalysisOutputFile> files = submission.getAnalysis()
+				.getAnalysisOutputFiles();
+		AnalysisOutputFile outputFile = null;
+		String htmlExt = "html";
+
+		for (AnalysisOutputFile file : files) {
+			if (file.getFile()
+					.toFile()
+					.getName()
+					.contains(filename) && FilenameUtils.getExtension(filename)
+					.equals(htmlExt)) {
+				outputFile = file;
+				break;
+			}
+		}
+
+		// Set the common Http headers
+		response.setHeader(HttpHeaders.CONTENT_DISPOSITION, "inline");
+		response.setHeader(HttpHeaders.CONTENT_TYPE, "text/html");
+
+		try (InputStream inputStream = new FileInputStream(outputFile.getFile()
+				.toString()); OutputStream outputStream = response.getOutputStream()) {
+			response.setHeader(HttpHeaders.CONTENT_LENGTH, String.valueOf(Files.size(outputFile.getFile())));
```

Comment:
I think the browser will handle things without this response header.  It looks like it should just stream the response.  That way if there's some filesystem size hiccup here it won't make the browser freak out that the content is different than the size it recieved.

Reply:
Sounds good I will remove that response header

Reply:
Updated in 1c116df5695ce8f3b89c8013f76a705e6dd3482f